knowledge/data-sources/website/colly.go

```diff
@@ -40,11 +44,31 @@ func crawlColly(ctx context.Context, input *MetadataInput, output *MetadataOutpu
 	}
 
 	output.Status = ""
+	output.State.WebsiteCrawlingState = WebsiteCrawlingState{
+		CurrentURL:  "",
+		VisitedURLs: nil,
+	}
 	return writeMetadata(ctx, output, gptscript)
 }
 
-func scrape(ctx context.Context, logOut *logrus.Logger, output *MetadataOutput, gptscriptClient *gptscript.GPTScript, visited map[string]struct{}, folders map[string]struct{}, url string, limit int) error {
+func scrape(ctx context.Context, logOut *logrus.Logger, output *MetadataOutput, gptscriptClient *gptscript.GPTScript, visited map[string]struct{}, url, urlToResume string, limit int) error {
 	collector := colly.NewCollector()
+
+	inMemoryStore := &storage.InMemoryStorage{}
```

Comment:
This is a manual way to inject visisted url into their default memory store so that it knows the URL visited from last try.